saw-remote-api/python/saw/llvm.py

```diff
@@ -23,6 +22,23 @@ def to_json(self) -> Any:
         server will dispatch on to then interpret the rest of the JSON object.``"""
         pass
 
+    @overload
+    def __getitem__(self, key : int) -> 'ElemVal':
+        pass
```

Comment:
Just to make sure I understand what this code is doing: all invocations of `setup_val[key]` will fall through to the second definition of `__getitem__`, right? In other words, this first definition of `__getitem__` with `@overload`/`pass` is purely for the benefit of typechecking? (It's a bit weird to have a function check `isinstance(key, int)` when the type signature has `key : str`, but if this is the Pythonic way to do this, I can't really complain too much.)

Reply:
Yes, that is my understanding and consistent with what I have observed thus far using this pattern.

From [PEP 484](https://www.python.org/dev/peps/pep-0484/#function-method-overloading): 

>Uses of the @overload decorator as shown above are suitable for stub files. In regular modules, a series of @overload-decorated definitions must be followed by exactly one non-@overload-decorated definition (for the same function/method). The @overload-decorated definitions are for the benefit of the type checker only, since they will be overwritten by the non-@overload-decorated definition, while the latter is used at runtime but should be ignored by a type checker.